cmd/drand-gossip-relay/main.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/tls"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/drand/drand/cmd/drand-gossip-relay/client"
+	"github.com/drand/drand/cmd/drand-gossip-relay/lp2p"
+	"github.com/drand/drand/protobuf/drand"
+	"github.com/golang/protobuf/proto"
+	"github.com/ipfs/go-datastore"
+	bds "github.com/ipfs/go-ds-badger2"
+	logging "github.com/ipfs/go-log/v2"
+	crypto "github.com/libp2p/go-libp2p-crypto"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	ma "github.com/multiformats/go-multiaddr"
+	cli "github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	log = logging.Logger("beacon-relay")
+)
+
+func main() {
+	logging.SetLogLevel("*", "info")
+	logging.SetLogLevel("beacon-relay", "info")
+
+	app := &cli.App{
+		Name:        "beacon-relay",
+		Version:     "0.0.1",
+		Description: "pubsub relay for randomness beacon",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "network-name",
+				Aliases: []string{"nn"},
+			},
+		},
+		Commands: []*cli.Command{runCmd, clientCmd},
+	}
+	err := app.Run(os.Args)
+	if err != nil {
+		fmt.Printf("error: %+v\n", err)
+		os.Exit(1)
+	}
+}
+func parseMultiaddrSlice(peer []string) ([]ma.Multiaddr, error) {
+	out := make([]ma.Multiaddr, len(peer))
+	for i, peer := range peer {
+		m, err := ma.NewMultiaddr(peer)
+		if err != nil {
+			return nil, xerrors.Errorf("parsing multiaddr\"%s\": %w", peer, err)
+		}
+		out[i] = m
+	}
+	return out, nil
+}
+
+var runCmd = &cli.Command{
+	Name: "run",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "connect",
+			Usage: "host:port to dial to a GRPC drand public API",
+		},
+		&cli.StringFlag{
+			Name:  "cert",
+			Usage: "file containing GRPC transport credentials of peer",
+		},
+		&cli.BoolFlag{
+			Name:  "insecure",
+			Usage: "allow insecure connection",
+		},
+		&cli.StringFlag{
+			Name:  "listen",
+			Usage: "listen addr for libp2p",
+			Value: "/ip4/0.0.0.0/tcp/44544",
+		},
+		&cli.StringSliceFlag{
+			Name:  "peer-with",
+			Usage: "list of peers to connect with",
+		},
```

Comment:
Private key file as arg?

Reply:
I can make private key file as an arg, but given that we already create datastore in current working directory I feel like it is redundant.

Private key bytes and argument (or any other secret) is very bad practice.